socorro/signature/rules.py

```diff
@@ -366,26 +367,37 @@ def generate(self, source_list, crashed_thread=None, delimiter=" | "):
         # have already been normalized at this point.
         new_signature_list = []
         for a_signature in source_list:
-            # If the signature matches the irrelevant signatures regex, skip to the next frame.
+            # If the frame signature is an empty string, skip it
+            if not a_signature:
+                debug_notes.append(f'empty string; ignoring: "{a_signature}"')
+                continue
+
+            # If the frame signature matches the irrelevant signatures regex, skip to
+            # the next frame
             if self.irrelevant_signature_re.match(a_signature):
                 debug_notes.append(f'irrelevant; ignoring: "{a_signature}"')
                 continue
 
-            # If the frame signature is a dll, remove the @xxxxx part.
-            if (
-                not a_signature.startswith("(unloaded")
-                and ".dll" in a_signature.lower()
+            # Skip repeating dll/so signatures
+            if not a_signature.startswith("(unloaded") and (
+                # A library on Windows or Linux
+                a_signature.lower().endswith((".dll", ".so"))
+                # XUL library on macOS
+                or a_signature == "XUL"
+                # Case where symbols file has "name omitted" as function symbol
+                or a_signature == "<name omitted>"
+                # Case where symbols file only has public function names. bug
+                # #1685178
+                or a_signature.startswith("<unknown in")
             ):
-                a_signature = a_signature.split("@")[0]
-
-                # If this trimmed DLL signature is the same as the previous frame's, skip it.
```

Comment:
Given that we're no longer including the module offset, we don't need to do this anymore.